core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -214,73 +204,75 @@ class UserService(protected val ctx: RawlsRequestContext,
-        (a: Set[SamResourceRole], b: Set[SamResourceRole]) => a ++ b
-      )
+      .groupBy(_.resourceId)
       .view
-      .mapValues { values: Set[SamResourceRole] => samRolesToProjectRoles(values) }
+      .mapValues(resources => samRolesToProjectRoles(resources.flatMap(r => r.direct.roles ++ r.inherited.roles).toSet))
       .toMap
     resourceIds = rolesByResourceId.keySet
     billingProfiles <- billingProfileManagerDAO.getAllBillingProfiles(ctx)
-    projectsInDB <- dataSource.inTransaction { dataAccess =>
-      dataAccess.rawlsBillingProjectQuery.getBillingProjects(resourceIds.map(RawlsBillingProjectName))
-    }
+    projectsInDB <- billingRepository.getBillingProjects(resourceIds.map(RawlsBillingProjectName))
     hardcodedBillingProject <- billingProfileManagerDAO.getHardcodedAzureBillingProject(resourceIds, ctx.userInfo)
-  } yield (projectsInDB ++ hardcodedBillingProject).toList.map { project =>
-    val roles = rolesByResourceId.getOrElse(project.projectName.value, Set())
-    val updatedProject = project.billingProfileId match {
-      case None => project
+  } yield (projectsInDB ++ hardcodedBillingProject).toList.map { p =>
+    val roles = rolesByResourceId.getOrElse(p.projectName.value, Set())
+    val billingProfile = p.billingProfileId.flatMap(id => billingProfiles.find(_.getId == UUID.fromString(id)))
+    mapCloudPlatform(p, billingProfile, roles)
+  }
+
+  /**
+    * Map the cloud platform to a billing project.
+    * if no BPM id is set it's a GCP project
+    * if a BPM id is set and a billing profile was found, it's Azure, and the coordinates are mapped (for now)
+    * if a BPM id is set and no billing profile was found, mark as UNKNOWN
+    */
+  def mapCloudPlatform(
+    project: RawlsBillingProject,
+    billingProfile: Option[ProfileModel],
+    roles: Set[ProjectRole]
+  ): RawlsBillingProjectResponse =
+    (project.billingProfileId, billingProfile) match {
+      case (None, _) => RawlsBillingProjectResponse(roles, project, CloudPlatform.GCP)
       // For projects in projectsInDB that have a billingProfileId, look up their managed app coordinates from BPM
-      case Some(id) =>
-        billingProfiles.find(_.getId == UUID.fromString(id)) match {
-          case Some(p) =>
-            val c = AzureManagedAppCoordinates(p.getTenantId, p.getSubscriptionId, p.getManagedResourceGroupId)
-            project.copy(azureManagedAppCoordinates = Some(c))
-          case None =>
-            project.copy(
-              message = Some(s"Unable to find billing profile in Billing Profile Manager for billing profile id: $id"),
-              status = CreationStatuses.Error
-            )
-        }
+      case (Some(_), Some(p)) =>
+        val c = AzureManagedAppCoordinates(p.getTenantId, p.getSubscriptionId, p.getManagedResourceGroupId)
+        RawlsBillingProjectResponse(roles, project.copy(azureManagedAppCoordinates = Some(c)), CloudPlatform.AZURE)
```

Comment:
@blakery thinking about this more, we probably should determine `AZURE` vs. `GCP` here based on the existence of the managed app coordinates.

Reply:
I'm not sure there's a difference, since the managed app coordinates are directly derived from the billing profile.

Reply:
I verified that the CloudPlatform is required in BPM when projects are created